### ISSUE TITLE
Segmentation fault on Windows GitHub-hosted runners with GDAL>3.6 and python<3.11

### ISSUE THREAD

**Issue by brynpickering** [OSGeo/gdal#8959]
## Expected behavior and actual behavior.

We are trying to update our GDAL version in https://github.com/arup-group/pam/ but hitting on a segmentation fault in the CI for Windows machines running on Python < 3.11 and GDAL >= 3.6. 

Here are the failing runs:
GDAL 3.7.3, py310: https://github.com/arup-group/pam/actions/runs/7182739378/job/19559929477
GDAL 3.8.1, py310: https://github.com/arup-group/pam/actions/runs/7180889878/job/19554117149
GDAL 3.8.1, py39: https://github.com/arup-group/pam/actions/runs/7179937434/job/19551761931

The traceback is completely useless but as I don't have a Windows machine I'm not sure I can get anything better. On running pytest, it simply states:

```sh
D:\a\_temp\fa47018d-70cc-403e-a6d7-19e0b81edf77.sh: line 1:   483 Segmentation fault      python -m pytest --no-cov
```

It isn't a problem for GDAL=3.5.3 (https://github.com/arup-group/pam/actions/runs/7182277134/job/19558422142) or 3.6.4 (https://github.com/arup-group/pam/actions/runs/7183112277/job/19561131871) or e.g. python=3.12 (https://github.com/arup-group/pam/actions/runs/7182739378/job/19559929171)

## Steps to reproduce the problem.

I would guess that trying this on a Windows machine locally will have the same result, so:

Clone the PAM repo and create mamba/conda env with updated GDAL pinning to >=3.6 (in `requirements/base.txt`) and `python=3.10`, then run the tests (`pytest`).

## Operating system

Microsoft Windows Server 2022 10.0.20348 (GitHub-hosted runner `windows-latest`)

## GDAL version and provenance

From the abovementioned logs:

3.7.3 (py310h66fcd94_12) - conda-forge
3.8.1 (py310h66fcd94_4) - conda-forge

**Comment by brynpickering** [comment]
Attaching logs here as they'll likely be deleted from our CI archive at some point:

[gdal_3_8_1-py_3_9.txt](https://github.com/OSGeo/gdal/files/13713879/gdal_3_8_1-py_3_9.txt)
[gdal_3_8_1-py_3_10.txt](https://github.com/OSGeo/gdal/files/13713896/gdal_3_8_1-py_3_10.txt)
[gdal_3_7_3-py_3_10.txt](https://github.com/OSGeo/gdal/files/13713905/gdal_3_7_3-py_3_10.txt)
[gdal_3_8_1-py_3_12.txt](https:/
[gdal_3_5_3-py_3_10.txt](https://github.com/OSGeo/gdal/files/13713923/gdal_3_5_3-py_3_10.txt)
/github.com/OSGeo/gdal/files/13713917/gdal_3_8_1-py_3_12.txt)

**Comment by schwehr** [comment]
Can you share a stack trace? Looking at the CI runs, I'm guessing you will need to change how the binary is build or windows is configured to force an actual stack trace to be reported. But, it's been > 2 decades since I really used Windows, so I'm guessing.

Maybe this is helpful?

https://stackoverflow.com/questions/77005/how-to-automatically-generate-a-stacktrace-when-my-program-crashes

**Comment by brynpickering** [comment]
OK, the issue seems to have resolved itself with no change on my side. Could be an update in the gdal build number or an update elsewhere. E.g., I noticed the build numbers in the visual studio C++ runtime library also updated in passing builds.

For future reference I tried to get the segementation fault traceback with `python -X faulthandler -m pytest -p no:faulthandler` (replacing the pytest handler with the base python one), although I can't say if it worked or not as tests are now passing.
